memory-viz/src/memory_model.ts

```diff
@@ -723,28 +754,33 @@ export class MemoryModel {
      *                        standard SVG attributes, documented on
      *                        https://developer.mozilla.org/en-US/docs/Web/SVG/Element/text.
      *                        For instance, {fill: 'blue', stroke: 'red'}
+     * @param {class}
      */
 
-    drawText(text, x, y, style) {
-        style["x"] = x;
-        style["y"] = y;
-
-        for (const style_attribute of Object.keys(default_text_style)) {
-            if (!style.hasOwnProperty(style_attribute)) {
-                style[style_attribute] = default_text_style[style_attribute];
-            }
-        }
-
+    drawText(text, x, y, style, text_class = null) {
         const newElement = this.document.createElementNS(
             "http://www.w3.org/2000/svg",
             "text"
         );
 
-        for (const style_attribute of Object.keys(style)) {
-            newElement.setAttribute(style_attribute, style[style_attribute]);
+        newElement.setAttribute("x", x);
+        newElement.setAttribute("y", y);
+
+        if (style !== undefined) {
+            let new_style = "";
+            for (const style_attribute of Object.keys(style)) {
+                new_style = new_style.concat(style_attribute);
```

Comment:
it's more idiomatic to use `+=` here; also, you can simplify by combining this with the next line, and using a single interpolated string for the whole expression